lib/service/kubernetes.go

```diff
@@ -218,6 +218,11 @@ func (process *TeleportProcess) initKubernetesService(log *logrus.Entry, conn *C
 		Streamer: streamer,
 	}
 
+	var publicAddr string
+	if len(cfg.Kube.PublicAddrs) > 0 {
```

Comment:
Under what circumstances would someone configure multiple public addresses? Is there any special handling we need to do in that case?

Reply:
Providing more than one address doesn't seem to be really supported. I think this was done this way to incorporate the` utils.Strings` parser and match other configuration fields for UX.

Only the first address provided is used for dialing/heart-beating, but all provided addresses are used as principals for kube TLS certificates. @awly is this what was intended, or should `kube_service` be updated to try connecting over all provided `public_addr`'s?

Either way I think adding logic for this special case is out of scope for this bug fix.

Reply:
I agree it's out of scope.
Generally, we support multiple `public_addrs` in services for TLS cert principals - a user may have several addresses for a service (e.g. one external-facing and one for internal networks, with separate firewall rules) and TLS certs should validate against all of them.

For internal routing using just the first address is OK for now, I haven't heard of anyone complaining.